hack/lib/start.sh

```diff
@@ -0,0 +1,576 @@
+#
+# Globals:
+#  - MASTER_CONFIG_DIR
+#  - SERVER_CONFIG_DIR
+#  - API_HOST
+#  - ETCD_PORT
+#  - ETCD_PEER_PORT
+#  - USE_SUDO
+#  - MAX_IMAGES_BULK_IMPORTED_PER_REPOSITORY
+# Returns:
+#  - export ADMIN_KUBECONFIG
+#  - export CLUSTER_ADMIN_CONTEXT
+function os::start::internal::patch_master_config() {
+	local sudo=${USE_SUDO:+sudo}
+	cp "${SERVER_CONFIG_DIR}/master/master-config.yaml" "${SERVER_CONFIG_DIR}/master/master-config.orig.yaml"
+	openshift ex config patch ${SERVER_CONFIG_DIR}/master/master-config.orig.yaml --patch="{\"etcdConfig\": {\"address\": \"${API_HOST}:${ETCD_PORT}\"}}" | \
+	openshift ex config patch - --patch="{\"etcdConfig\": {\"servingInfo\": {\"bindAddress\": \"${API_HOST}:${ETCD_PORT}\"}}}" | \
+	openshift ex config patch - --type json --patch="[{\"op\": \"replace\", \"path\": \"/etcdClientInfo/urls\", \"value\": [\"${API_SCHEME}://${API_HOST}:${ETCD_PORT}\"]}]" | \
+	openshift ex config patch - --patch="{\"etcdConfig\": {\"peerAddress\": \"${API_HOST}:${ETCD_PEER_PORT}\"}}" | \
+	openshift ex config patch - --patch="{\"etcdConfig\": {\"peerServingInfo\": {\"bindAddress\": \"${API_HOST}:${ETCD_PEER_PORT}\"}}}" | \
+	openshift ex config patch - --patch="{\"imagePolicyConfig\": {\"maxImagesBulkImportedPerRepository\": ${MAX_IMAGES_BULK_IMPORTED_PER_REPOSITORY:-5}}}" > "${SERVER_CONFIG_DIR}/master/master-config.yaml"
+
+	# Make oc use ${MASTER_CONFIG_DIR}/admin.kubeconfig, and ignore anything in the running user's $HOME dir
+	export ADMIN_KUBECONFIG="${MASTER_CONFIG_DIR}/admin.kubeconfig"
+	CLUSTER_ADMIN_CONTEXT=$(oc config view --config=${ADMIN_KUBECONFIG} --flatten -o template --template='{{index . "current-context"}}'); export CLUSTER_ADMIN_CONTEXT
+	${sudo} chmod -R a+rwX "${ADMIN_KUBECONFIG}"
+	echo "[INFO] To debug: export KUBECONFIG=$ADMIN_KUBECONFIG"
+}
+readonly -f os::start::internal::patch_master_config
+
+# os::start::server starts the OpenShift server, exports the PID of the OpenShift server and waits until openshift server endpoints are available
+# It is advised to use this function after a successful run of 'configure_os_server'
+#
+# Globals:
+#  - USE_SUDO
+#  - LOG_DIR
+#  - ARTIFACT_DIR
+#  - VOLUME_DIR
+#  - SERVER_CONFIG_DIR
+#  - USE_IMAGES
+#  - MASTER_ADDR
+#  - MASTER_CONFIG_DIR
+#  - NODE_CONFIG_DIR
+#  - API_SCHEME
+#  - API_HOST
+#  - API_PORT
+#  - KUBELET_SCHEME
+#  - KUBELET_HOST
+#  - KUBELET_PORT
+# Arguments:
+#  1 - API server version (ie. "v1.2.0")
+#  2 - Controllers version (ie. "v1.2.0")
+#  3 - Skip node start ("1" to skip node start)
+# Returns:
+#  - export OS_PID
+#  - export ETCD_PID
+#  - export API_SERVER_PID
+#  - export CONTROLLERS_PID
+#  - export NODE_PID
+function os::start::server() {
+	local api_server_version="${1:-}"
+	local controllers_version="${2:-}"
+	local skip_node="${3:-}"
+
+	echo "[INFO] Scan of OpenShift related processes already up via ps -ef	| grep openshift : "
+	ps -ef | grep openshift
```

Comment:
@csrwng I see, you should be able to do:
```sh
command <(subshell)
```

In that case instead of 
```sh
command <<< "$(subshell)"
```

But that's not great that it doesn't work